### PR TITLE
Wmedders21/appeals 64703

### PR DESCRIPTION
**Pull request by wmedders21**
Resolves [APPEALS-64703](https://jira.devops.va.gov/browse/APPEALS-64703)

# Description
Please explain the changes you made here.
- Add the functionality for the submit button in the CancelReviewTranscriptTaskModal to make a patch request to the backend to cancel the task
- Adds a patch route on the back end to handle this request
- Adds a controller action with the logic to cancel the task and update its task instructions for display on the case timeline.

## Acceptance Criteria
- [ ] Code compiles correctly

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
- Log in to caseflow as a Hearing Admin
- Visit the case details page for an appeal that has an assigned ReviewTranscriptTask for the logged-in user
  - `/queue/appeals/<appeal-uuid>`
- Select "Cancel task" from the task actions dropdown
- Fill out the text box in the CancelReviewTranscriptTaskModal
- Click on "Cancel task" to submit
- Expect to be on the Case details page
- Expect that the ReviewTranscriptTask shows up as cancelled on the Case Timeline with instructions as described in the jira ticket.

- [ ] For feature branches merging into main: Was this deployed to UAT?

# Frontend
## User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

# Backend
## Database Changes
*Only for Schema Changes*

* [ ] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ ] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/main/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] For queries using raw sql was an explain plan run by System Team
* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [ ] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

## Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
  * [ ] Check that calls are wrapped in MetricService record block
* [ ] Check that all configuration is coming from ENV variables
  * [ ] Listed all new ENV variables in description
  * [ ] Worked with or notified System Team that new ENV variables need to be set
* [ ] Update Fakes
* [ ] For feature branches: Was this tested in Caseflow UAT

# Best practices
## Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [ ] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [ ] No new code climate issues added

### Monitoring
- [ ] Are performance metrics (e.g., response time, throughput) being tracked?
- [ ] Are key application components monitored (e.g., database, cache, queues)?
- [ ] Is there a system in place for setting up alerts based on performance thresholds?

### Logging
- [ ] Are logs being produced at appropriate log levels (debug, info, warn, error, fatal)?
- [ ] Are logs structured (e.g., using log tags) for easier querying and analysis?
- [ ] Are sensitive data (e.g., passwords, tokens) redacted or omitted from logs?
- [ ] Is log retention and rotation configured correctly?
- [ ] Are logs being forwarded to a centralized logging system if needed?

### Auditing
- [ ] Are user actions being logged for audit purposes?
- [ ] Are changes to critical data being tracked ?
- [ ] Are logs being securely stored and protected from tampering or exposing protected data?

### Error Handling
- [ ] Are errors being caught and handled gracefully?
- [ ] Are appropriate error messages being displayed to users?
- [ ] Are critical errors being reported to an error tracking system (e.g., Sentry, ELK)?
- [ ] Are unhandled exceptions being caught at the application level ?

### Exception Handling
- [ ] Are custom exceptions defined and used where appropriate?
- [ ] Is exception handling consistent throughout the codebase?
- [ ] Are exceptions logged with relevant context and stack trace information?
- [ ] Are exceptions being grouped and categorized for easier analysis and resolution?

### CODE REVIEW

**Review by mchbidwell — COMMENTED** [review]
There are some test failures in the CancelReviewTranscriptTaskModal tests, please take a look.

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -194,14 +194,14 @@ def upload_transcription_to_vbms
...
       document_params =
-      {
-        veteran_file_number: appeal.veteran_file_number,
-        document_type: "Hearing Transcript",
-        document_subject: "notifications",
-        document_name: file_name,
-        application: "notification-report",
-        file: current_file_path
-      }
+        {
```
Comment (ThorntonMatthew) [inline app/controllers/tasks_controller.rb:197]: Rubocop fix

File: app/controllers/tasks_controller.rb [hunk 2]
```diff
@@ -229,6 +229,7 @@ def complete_transcript_review
       closed_at: Time.zone.now,
       completed_by_id: current_user.id
     )
+  end
```
Comment (ThorntonMatthew) [inline app/controllers/tasks_controller.rb:232]: This was the missing `end`.

File: app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -244,8 +245,6 @@ def cancel_review_transcript_task
     render_update_errors(error)
   end
 
-  private
```
Comment (ThorntonMatthew) [inline app/controllers/tasks_controller.rb]: Rubocop fix:

There is a `private` keyword on line 215 already.

File: app/controllers/tasks_controller.rb (hunk 2 above)
Comment (ThorntonMatthew) [reply app/controllers/tasks_controller.rb:232]: ```[127.0.0.1] [db133955-1eb5-445e-852b-22f78555374c] [localhost] SyntaxError (/__w/caseflow/caseflow/app/controllers/tasks_controller.rb:466: syntax error, unexpected end-of-input, expecting `end'):
[127.0.0.1] [db133955-1eb5-445e-852b-22f78555374c] [localhost]   
[127.0.0.1] [db133955-1eb5-445e-852b-22f78555374c] [localhost] app/controllers/tasks_controller.rb:466: syntax error, unexpected end-of-input, expecting `end'
[127.0.0.1] [db133955-1eb5-445e-852b-22f78555374c] [localhost] app/controllers/tasks_controller.rb:466: syntax error, unexpected end-of-input, expecting `end'
```
That was the error in the test logs.

**Review by mchbidwell — APPROVED** [review]

**Pull request merged**